atc/policy/opa/result.go

```diff
@@ -0,0 +1,92 @@
+package opa
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type opaResult struct {
+	allowed     bool
+	shouldBlock bool
```

Comment:
By default, a policy violation should block. If you just set a policy `default decision = {"allowed": false}`, it should block. This PR is introducing an ability of "not blocking". 

The `reasons` -> `messages` change doesn't break anything. `ResultMessageKey`'s default value is `result.reasons`.

Reply:
I think there was some way of setting `decision = {"allowed": false, "block": null}` so that it would parse it into `v == nil`, which skips the check and causes `shouldBlock` to use the default value of `false`. But this is irrelevant with the new way of parsing the result.

https://github.com/concourse/concourse/blob/9226d60146b056603230ea696a4cde26095381de/atc/policy/opa/result.go#L51-L53